staging/src/k8s.io/apimachinery/pkg/util/wait/wait_test.go

```diff
@@ -48,6 +49,26 @@ func TestUntil(t *testing.T) {
 	<-called
 }
 
+func TestUntilWithContext(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	cancel()
+	UntilWithContext(ctx, func(context.Context) {
+		t.Fatal("should not have been invoked")
+	}, 0)
+
+	ctx, cancel = context.WithCancel(context.TODO())
+	called := make(chan struct{})
+	go func() {
+		UntilWithContext(ctx, func(context.Context) {
+			called <- struct{}{}
+		}, 0)
+		close(called)
+	}()
+	<-called
+	cancel()
+	<-called
```

Comment:
I know you copied this line from existing tests, but what does this line verify?

Reply:
I indeed took it from the other test. I think it is just a sanity check to ensure that the channel is closed.

Reply:
But if the `called <- struct{}{}` is still being executing, then this line will still pass, right?

Reply:
I would agree on that, yes. Then maybe another option is that it is trying to prevent leaving an (internal) goroutine stuck on a send which may have happened right before the stop/cancel.

Reply:
@caesarxuchao as an outcome of this discussion, do you want me to change anything here?